backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleExecutor.java

```diff
@@ -140,90 +141,51 @@ public AnsibleReturnValue runCommand(AnsibleCommandConfig command,
                 return ret;
             }
 
-            // Process the events of the playbook:
-            while (iteration < timeout * 60) {
-                Thread.sleep(POLL_INTERVAL);
-
-                // Get the current status of the playbook:
-                AnsibleRunnerHttpClient.PlaybookStatus playbookStatus = runnerClient.getPlaybookStatus(playUuid);
-                String status = playbookStatus.getStatus();
-                msg = playbookStatus.getMsg();
-                // Process the events if the playbook is running:
-                totalEvents = runnerClient.getTotalEvents(playUuid);
-
-                if (
-                        msg.equalsIgnoreCase("running")
-                                || msg.equalsIgnoreCase("successful") && lastEventId < totalEvents
-                ) {
-                    lastEventId = runnerClient.processEvents(playUuid, lastEventId, fn, msg, ret.getLogFile());
-                    iteration += POLL_INTERVAL / 1000;
-                } else if (msg.equalsIgnoreCase("successful")) {
-                    // Exit the processing if playbook finished:
-                    ret.setAnsibleReturnCode(AnsibleReturnCode.OK);
-                    return ret;
-                } else if (status.equalsIgnoreCase("unknown")) {
-                    // ignore and continue:
-                } else if (runnerClient.isHostUnreachable(playUuid)) {
-                    ret.setStderr(msg);
-                    ret.setAnsibleReturnCode(AnsibleReturnCode.UNREACHABLE);
-                    return ret;
-                } else {
-                    // Playbook failed:
-                    return ret;
-                }
-            }
-
-            // Cancel playbook, and raise exception in case timeout occur:
-            runnerClient.cancelPlaybook(playUuid);
-            throw new PlaybookExecutionException(
-                "Timeout exceed while waiting for playbook. ", command.playbook()
-            );
+            ret = runnerClient.artifactHandler(commandConfig.getUuid(), lastEventId, timeout, fn);
         } catch (InventoryException ex) {
+            commandConfig.cleanup();
             String message = ex.getMessage();
             log.error("Error executing playbook: {}", message);
             log.debug("InventoryException: ", ex);
             ret.setStderr(message);
             ret.setAnsibleReturnCode(message.contains(SSH_TIMEOUT) ? AnsibleReturnCode.UNREACHABLE: AnsibleReturnCode.FAIL);
         } catch (Exception ex) {
+            commandConfig.cleanup();
             log.error("Exception: {}", ex.getMessage());
             log.debug("Exception: ", ex);
             ret.setStderr(ex.getMessage());
         } finally {
+            commandConfig.cleanup();
             // Make sure all events are proccessed even in case of failure:
-            if (playUuid != null && runnerClient != null && !async) {
+            if (playUuid != null && runnerClient != null && !async) { //TODO
```

Comment:
What missing in empty TODO?

Reply:
nothing, it was so that I won't forget the clean up, 
removed it